backend/capellacollab/core/authentication/injectables.py

```diff
@@ -3,19 +3,95 @@
 
 from __future__ import annotations
 
+import dataclasses
+import logging
+
 import fastapi
 from fastapi import status
+from fastapi.openapi import models as openapi_models
+from fastapi.security import base as security_base
 
 from capellacollab.core import database
-from capellacollab.core.authentication import jwt_bearer
+from capellacollab.core.authentication import basic_auth, jwt_bearer
 from capellacollab.projects import crud as projects_crud
 from capellacollab.projects import models as projects_models
 from capellacollab.projects.users import crud as projects_users_crud
 from capellacollab.projects.users import models as projects_users_models
 from capellacollab.users import crud as users_crud
 from capellacollab.users import models as users_models
 
-from . import helper
+logger = logging.getLogger(__name__)
+
+
+class OpenAPIFakeBase(security_base.SecurityBase):
+    """Fake class to display the authentication methods in the OpenAPI docs
+
+    fastAPI uses DependencyInjection together with the SecurityBase class
+    to determine which authentication methods are available.
+    More information in fastapi/dependencies/utils::get_sub_dependant
+    """
+
+    def __init__(self) -> None:
+        pass
+
+    def __call__(self) -> None:
+        pass
+
+    def __hash__(self) -> int:
+        return hash(self.__class__.__name__)
+
+
+@dataclasses.dataclass()
+class OpenAPIPersonalAccessToken(OpenAPIFakeBase):
+    """Displays the personal access token as authentication method in the OpenAPI docs"""
+
+    model = openapi_models.HTTPBase(
+        scheme="basic",
+        description="Can be used to authenticate with an personal access token",
+    )
+    scheme_name = "PersonalAccessToken"
+
+    __hash__ = OpenAPIFakeBase.__hash__
+
+
+@dataclasses.dataclass()
+class OpenAPIBearerToken(OpenAPIFakeBase):
+    """Displays the JWT Bearer token as authentication method in the OpenAPI docs"""
+
+    model = openapi_models.HTTPBase(
+        scheme="bearer",
+    )
+    scheme_name = "JWTBearer"
+
+    __hash__ = OpenAPIFakeBase.__hash__
+
+
+async def get_username(
+    request: fastapi.Request,
+    _unused1=fastapi.Depends(OpenAPIPersonalAccessToken()),
+    _unused2=fastapi.Depends(OpenAPIBearerToken()),
+):
+    basic = await basic_auth.HTTPBasicAuth(auto_error=False)(request)
+    jwt = await jwt_bearer.JWTBearer(auto_error=False)(request)
+
+    jwt_username, jwt_error = jwt
+    if jwt_username:
+        return jwt_username
+
+    basic_username, basic_error = basic
+    if basic_username:
+        return basic_username
+
+    if jwt_error:
+        raise jwt_error
+    if basic_error:
+        raise basic_error
```

Comment:
```suggestion
```suggestion
    jwt_username, jwt_error = jwt
    basic_username, basic_error = basic
    
    if username := jwt_username or basic_username:
        return username
    if error := jwt_error or basic_error:
        raise error
```
```